### DISCUSSION TITLE
Mixcore v2.0 has been switched to [develop] brand

### DISCUSSION THREAD

**Discussion by Smilefounder** [mixcore/mix.core#587] [Announcements]
Hi Forks,

Please be informed that Mixcore v2.0 has been switched to [develop] brand. As Mixcore v2 has been changed a lot, so please be aware that you should not use this v2 to upgrade from v1.

Git commit:
https://github.com/mixcore/mix.core/commit/811d027943007df600f94e360651ac67cf1270fa

```
git clone --branch develop --recursive https://github.com/mixcore/mix.core.git
git submodule update --init --recursive
```

Best regards,
Mixcore Team.

**Comment by furkandg** [comment]
Hello, I am getting an error while installing.

![image](https://user-images.githubusercontent.com/80524882/155485800-370afb3c-3183-47dd-a9d0-314ceaf590c6.png)

**Reply by Smilefounder** [reply]
@furkandg please delete :
mix.core/src/Mix.Cms.Web/MixContent

Then try again.

**Reply by furkandg** [reply]
My screen stays like this after deleting

![image](https://user-images.githubusercontent.com/80524882/155489770-6b7c333d-2de4-4dab-8957-29f0a2353ffe.png)

**Reply by Smilefounder** [reply]
@nhathoang989 kindly advice on this issue.

**Reply by furkandg** [reply]
What could cause this problem?

**Reply by Smilefounder** [reply]
@furkandg are you able to screen capture the browser debug logs? We can have more idea on the issue detail.

**Reply by furkandg** [reply]
It does not give any error while debugging.

**Reply by nhathoang989** [reply]
Hi @furkandg, I think you should check your database. Please make sure that the database is empty before setting up a new site.

**Reply by furkandg** [reply]
I just opened a new database for this operation

**Reply by Smilefounder** [reply]
@furkandg are you able to share with us the ftp or working environment?
Btw, you are using develop or master branch?

**Reply by Smilefounder** [reply]
Hi @furkandg 
We just found out that I forget to mentioned about you should update git submodule as well. Please try again then let us know the result.

```
git clone --branch develop --recursive https://github.com/mixcore/mix.core.git
git submodule update --init --recursive
```

**Reply by moalamri** [reply]
Hi @Smilefounder what’s the different between develop and develop-2 branches?

**Reply by moalamri** [reply]
By the way I get the same issue as @furkandg in version 1.0.5 only when Postgres is selected.
I haven’t tested v2 yet. I’ll be updating here?

**Reply by Smilefounder** [reply]
> Hi @Smilefounder what’s the different between develop and develop-2 branches?

Develop-2 branch will be deleted soon. Sorry for the confusing.

**Reply by Smilefounder** [reply]
> updating

Please check if you are pointing to the existing db or totally new db? Please try again with new db and delete application.json file before try again.

**Reply by moalamri** [reply]
A totally new Db.
I have never used mixcore so I think it’s better for me to start with the latest release without trying to fix the previous one?
I’ll try with the new version and let you know if I encountered something.

**Reply by moalamri** [reply]
The develop version is asking me to install Net Framework 4.0 to use MSBuild.
It happens either with `dotnet` cmd or Visual Studio 2022 IDE
`Mixcore.sln : Solution file error MSB4249: Unable to build website project "shared". The ASP.NET compiler is only available on the .NET Framework version of MSBuild.`

**Reply by moalamri** [reply]
Postgres is working good with the new release aka 2.
But there's a problem when stepping from 2 to 3
It just keep spinning and throw this in console (Happens with all DBs)
```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'apiEncryptKey')
    at Object._decryptAES [as decryptAES] (app-shared.min.js:2043:58)
    at _fillAuthData (app-shared.min.js:1076:25)
    at Object._sendRequest [as sendRequest] (app-shared.min.js:1183:37)
    at Object._register [as register] (app-init.min.js:300:31)
    at t.$scope.register (app-init.min.js:214:39)
    at fn (eval at compile (framework.min.js:1:122083), <anonymous>:4:144)
    at r (framework.min.js:1:94151)
    at t.$eval (framework.min.js:1:71887)
    at t.$apply (framework.min.js:1:72130)
    at HTMLFormElement.<anonymous> (framework.min.js:1:94236)
```
I guess it's something about appSettings in localStorage.

**Reply by nhathoang989** [reply]
Hi @moalamri 
There are some issue with Postgres migration and entity configurations. they're fixed in this commit https://github.com/mixcore/mix.core/commit/5faef8650f3fb9f10d1b5ba7445340a8859a5081.

**Reply by moalamri** [reply]
Hello @nhathoang989 
Thank you for fixing this. I know why I got the same error with other db option: sqlite.
After hard-refresh and cache clear to the browser most of console errors are gone (because I used the old version on the same IP) perhaps you could append version number to assets files to avoid conflicting?
I will try the new commit and give you feedback :).

**Comment by moalamri** [comment]
Hi @nhathoang989 
I used this commit [5faef86](https://github.com/mixcore/mix.core/commit/5faef8650f3fb9f10d1b5ba7445340a8859a5081) & [0615596](https://github.com/mixcore/mix.core/commit/0615596900046a84fe1daa9a7ae76fb4fba1ff5c) they both gave those errors when going to step 3:
<img width="401" alt="Screenshot" src="https://user-images.githubusercontent.com/964077/155870272-ed9c8e39-2e2a-4130-af28-12e13d0429a1.png">

**Reply by moalamri** [reply]
Please check this [Mixcore.sln#L50](https://github.com/mixcore/mix.core/blob/0615596900046a84fe1daa9a7ae76fb4fba1ff5c/Mixcore.sln#L49)

Visual Studio remove it on launch because of
`Mixcore.sln : Solution file error MSB4249: Unable to build website project "shared". The ASP.NET compiler is only available on the .NET Framework version of MSBuild.`

**Comment by nhathoang989** [comment]
hi @moalamri ,
You need to update submodule (mix.heart), then remove folder "MixContent" and "wwwroot/mix-content" from src/applications/Mixcore and reinstall again.

![image](https://user-images.githubusercontent.com/5327699/155871300-98286f45-cdd8-4c27-989c-d196f2ca0a08.png)

**Reply by moalamri** [reply]
@nhathoang989 
Hey,
Yes it is already the latest `Submodule path 'src/framework/mix-heart': checked out '9626ea0cc92caa3058929b738b6a946381e51169'`

Do you have Discord channel so we can look into this faster :)?
Thank you

**Reply by develite98** [reply]
@moalamri Please join our Discord server https://discord.gg/3cBzybJH for more discussion.

**Reply by nhathoang989** [reply]
Hi @moalamri , I can reproduce your issue by clone to a new folder.
Please pull latest from develop and try again
https://github.com/mixcore/mix.core/commit/8eddfb136f7f02d4bba8d509025ed1abb23b9e50

**Reply by furkandg** [reply]
My problem was solved when I changed the database. Error over Postgresql

**Reply by moalamri** [reply]
@furkandg Could you please confirm if this issue happens with v2 in this commit [8eddfb1](https://github.com/mixcore/mix.core/commit/8eddfb136f7f02d4bba8d509025ed1abb23b9e50)
